src/Translation_unit.ml

```diff
@@ -137,169 +121,39 @@ let ellipsis n msg =
 
 let ellipsis_cmt = ellipsis 50
 
-let parse_print (XUnit xunit) (conf : Conf.t) ~input_name ~input_file ic
-    ofile =
+let with_file input_name output_file suf ext f =
   let dir =
-    match ofile with
-    | Some ofile -> Filename.dirname ofile
+    match output_file with
+    | Some filename -> Filename.dirname filename
     | None -> Filename.get_temp_dir_name ()
   in
-  let base = Filename.(remove_extension (basename input_name)) in
+  let base = Filename.remove_extension (Filename.basename input_name) in
+  let tmp = Filename.concat dir (base ^ suf ^ ext) in
+  Out_channel.with_file tmp ~f ;
+  tmp
+
+let dump_ast ~input_name ?output_file ~suffix fmt =
+  if Conf.debug then
+    let ext = ".ast" in
+    let (_filename : string) =
+      with_file input_name output_file suffix ext (fun oc ->
+          fmt (Caml.Format.formatter_of_out_channel oc) )
+    in
+    ()
```

Comment:
How about `with_file ... |> (ignore : string -> _)` instead of `let ... = ... in ()`?